nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/service/nifi-common.service.ts

```diff
@@ -171,4 +187,230 @@ export class NiFiCommon {
+        duration = Math.floor(duration % NiFiCommon.MILLIS_PER_SECOND);
+
+        // format the time
+        const time =
+            this.pad(hours, 2, '0') +
+            ':' +
+            this.pad(minutes, 2, '0') +
+            ':' +
+            this.pad(seconds, 2, '0') +
+            '.' +
+            this.pad(duration, 3, '0');
+
+        // only include days if appropriate
+        if (days > 0) {
+            return days + ' days and ' + time;
+        } else {
+            return time;
+        }
+    }
+
+    /**
+     * Formats the specified number of bytes into a human readable string.
+     *
+     * @param {number} dataSize
+     * @returns {string}
+     */
+    public formatDataSize(dataSize: number): string {
+        let dataSizeToFormat: number = parseFloat(`${dataSize / NiFiCommon.BYTES_IN_TERABYTE}`);
+        if (dataSizeToFormat > 1) {
+            return dataSizeToFormat.toFixed(2) + ' TB';
+        }
+
+        // check gigabytes
+        dataSizeToFormat = parseFloat(`${dataSize / NiFiCommon.BYTES_IN_GIGABYTE}`);
+        if (dataSizeToFormat > 1) {
+            return dataSizeToFormat.toFixed(2) + ' GB';
+        }
+
+        // check megabytes
+        dataSizeToFormat = parseFloat(`${dataSize / NiFiCommon.BYTES_IN_MEGABYTE}`);
+        if (dataSizeToFormat > 1) {
+            return dataSizeToFormat.toFixed(2) + ' MB';
+        }
+
+        // check kilobytes
+        dataSizeToFormat = parseFloat(`${dataSize / NiFiCommon.BYTES_IN_KILOBYTE}`);
+        if (dataSizeToFormat > 1) {
+            return dataSizeToFormat.toFixed(2) + ' KB';
+        }
+
+        // default to bytes
+        return parseFloat(`${dataSize}`).toFixed(2) + ' bytes';
+    }
+
+    /**
+     * Formats the specified integer as a string (adding commas). At this
+     * point this does not take into account any locales.
+     *
+     * @param {integer} integer
+     */
+    public formatInteger(integer: number): string {
+        let string: string = integer + '';
+        const regex: RegExp = /(\d+)(\d{3})/;
+        while (regex.test(string)) {
+            string = string.replace(regex, '$1' + ',' + '$2');
+        }
+        return this.escapeHtml(string);
+    }
+
+    /**
+     * Formats the specified float using two decimal places.
+     *
+     * @param {float} f
+     */
+    public formatFloat(f: number): string {
+        if (!f) {
+            return 0.0 + '';
+        }
+        return f.toFixed(2) + '';
+    }
+
+    private static readonly entityMap: { [key: string]: string } = {
+        '&': '&amp;',
+        '<': '&lt;',
+        '>': '&gt;',
+        '"': '&quot;',
+        "'": '&#39;',
+        '/': '&#x2f;'
+    };
+
+    /**
+     * HTML escapes the specified string. If the string is null
+     * or undefined, an empty string is returned.
+     *
+     * @returns {string}
+     */
+    public escapeHtml(string: string): string {
```

Comment:
In the current implementation of the front end, we needed to be very careful with input especially when used in conjunction with `$(...).append()` and `$(...).html()`. Since Angular provides this sanitization by default when interpolating content, do we still need it here?

Reply:
Yeah, probably not needed. It is only used (in this PR) in the `formatInteger` function that takes in a number, so it can never contain invalid chars anyway. I'll remove it.